packages/react/react/src/hooks/useTimeout.ts

```diff
@@ -1,24 +1,14 @@
-import { useEffect, useRef } from 'react';
+import { noop } from '@toss/utils';
+import { useEffect } from 'react';
+import { usePreservedCallback } from './usePreservedCallback';
 
 /** @tossdocs-ignore */
 export function useTimeout(callback: () => void, delay = 0) {
-  const savedCallback = useRef<() => void>();
+  const savedCallback = usePreservedCallback(callback ?? noop);
```

Comment:
if we set the default value like this, we don't need to use conditional statements to determine whether the callback function exists or not.

```tsx
function handleTimeout() {
  // (*)
  if (savedCallback.current) {
    savedCallback.current();
  }
}
```

I think it would make the code more readable!! 💪